lib/backend/k8s/syncer.go

```diff
@@ -174,16 +218,42 @@ func (syn *kubeSyncer) readFromKubernetesAPI() {
 			// Event is OK - parse it and send it over the channel.
 			upd = syn.parseNetworkPolicyEvent(event)
 			latestVersions.networkPolicyVersion = upd.KVPair.Revision.(string)
-			syn.callbacks.OnUpdates([]api.Update{*upd})
+			syn.sendUpdates([]api.Update{*upd})
 		}
 	}
 }
 
+func (syn *kubeSyncer) getUpdateTypeShapshot(kvp model.KVPair) api.UpdateType {
+	if _, ok := syn.tracker[kvp.Key]; ok {
+		return api.UpdateTypeKVNew
+	} else {
+		return api.UpdateTypeKVUpdated
+	}
+}
+
+func (syn *kubeSyncer) performSnapshotDeletes(exists map[model.Key]bool) {
+	log.Info("Checking for any deletes for snapshot")
+	deletes := []api.Update{}
+	for cachedKey, _ := range syn.tracker {
+		// Check each cached key to see if it exists in the snapshot.  If it doesn't,
+		// we need to send a delete for it.
```

Comment:
The way I've got it right now, that always occurs in `sendUpdates`. 

Tried to push all the tracker modification / sending of updates through one place so that the behavior is always the same.  

This method just builds the set of delete updates to send.